staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -527,7 +527,19 @@ func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion
 				return result, w.resourceVersion, matchValue.IndexName, nil
 			}
 		}
-		return w.store.List(), w.resourceVersion, "", nil
+
+		allItems := w.store.List()
```

Comment:
Please don't do that under lock to avoid lock contention.

Reply:
good point, it wasn't clear because the lock is acquired by `waitUntilFreshAndBlock` method.